### PR TITLE
Financroo aggregate accounts, transactions and balances

### PR DESCRIPTION
**Pull request by ikawalec**
- define banks and login client credentials in config.yaml instead of env variables
- store connected banks in a mounted key-value store
- aggregate accounts, transactions and balances based on connected banks
- replace ACP_HOST, ACP_INTERNAL_HOST with ACP_URL, ACP_INTERNAL_URL
- add disconnect bank button
- reconnect to bank when refresh token expires

### CODE REVIEW

**Review by bwereszczak — APPROVED** [review]

File: financroo/bank_aggregation_routes.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		}
+
+		// todo parallel
+		for i, b := range user.Banks {
+			if client, token, err = s.GetClientWithToken(b); err != nil {
+				c.String(http.StatusUnauthorized, err.Error())
+				return
+			}
+			if err = s.UserRepo.Set(user); err != nil {
+				c.String(http.StatusInternalServerError, fmt.Sprintf("failed to update user: %+v", err))
```
Comment (mbilski) [inline financroo/bank_aggregation_routes.go:94]: missing return

File: financroo/bank_aggregation_routes.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
```
Comment (mbilski) [inline financroo/bank_aggregation_routes.go:146]: same here

File: financroo/bank_aggregation_routes.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
```
Comment (mbilski) [inline financroo/bank_aggregation_routes.go:197]: same here

File: financroo/data/config.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+login:
+  tenant_id: default
+  server_id: financroo
+  client_id: bv3p4uo557ouq8rha2bg
+  ui_url: https://${APP_HOST}:8091
```
Comment (mbilski) [inline financroo/data/config.yaml:5]: I don't like it that we need to sed this config file

**Review by mbilski — CHANGES_REQUESTED** [review]

File: financroo/config.go [hunk 5]
```diff
@@ -3,35 +3,113 @@ package main
...
+		config = Config{}
+		bs     []byte
+		err    error
+	)
+
+	if err = env.Parse(&config); err != nil {
+		return config, err
+	}
+
+	if bs, err = ioutil.ReadFile(config.YAMLConfigFile); err != nil {
```
Comment (mbilski) [inline financroo/config.go:75]: wouldn't it be better to use viper to manage config files and env var instead of reading yaml file manually?

Comment (ikawalec) [reply financroo/config.go:75]: If viper can read, validate if the env is provided, and also set the default env value I can replace it.

Comment (ikawalec) [reply financroo/config.go]: I've changed it to use vuper. Please take a look.

**Review by mbilski — APPROVED** [review]

**Pull request merged**